docs/development/pigs/pig-016.rst

```diff
@@ -0,0 +1,215 @@
+- gammapy/time/models -- gammapy/modeling/models/time
+
+Since it is very inconvenient for users to have to type sub-sub-sub-package
+names, we propose to expose the end-user API direction in gammapy.modeling for
+general classes (Parameter, Parameters, Model, Fit, Dataset). For the models, we
+propose to expose them in gammapy.modeling.models (same as Astropy), and to
+introduce a factory function to make it easy to make models that is easy to
+import and use. E.g. it could be gammapy.modeling.make_model and take a model
+specification string or dictionary.
+
+We propose to add "Source" to all (currently 3D) source models, and to add
+"Spectral", "Spatial" and "Time" for the spectral, spatial and time models. This
+avoids user confusion concerning the type of model, which currently exists in
+v0.13, because for some models (Gaussian, diffuse, table, constant) it isn't
+obvious what kind they are.
+
+The motivation for this change is weak, but in discussions at the Erlangen
+coding sprint most Gammapy developers preferred it. The reasoning was that it's
+a bit easier to find and maintain models if they are all in one place.
+
+There was a concern that this might introduce circular dependencies between
+gammapy.modeling.models and e.g. gammapy.spectrum, gammapy.image, gammapy.cube
+or gammapy.time. But in `GH 2290`_ it was shown that this is not the case, the
+proposed change works well.
+
+There is still the open question whether models will keep this simple interface,
+or if we will add an evaluation caching layer, or whether eventually reduced
+IRFs will become models. So it is very much possible that there will be other
+changes concerning the Gammapy package structure in the future.
+
+Dissolve gammapy.background
+---------------------------
+
+We propose to dissolve gammapy.background, and move the code like this:
+
+* background_estimate.py, reflected.py, phase.py -> gammapy/spectrum
+* ring.py -> gammapy/cube/background_ring.py
+
+The motivation for this change is that currently map-related background
+estimation code is split into gammapy/background and gammapy/cube without a
+clear separation what goes where. So the choice was either to move everything to
+dissolve gammapy.background or to move all background-related code to
+gammapy.background.
+
+Looking at the package structure illustration above, it's clear that
+gammapy.background is problematic, because it is lower-level than
+gammapy.spectrum and gammapy.maps, so moving background estimation classes from
+gammapy.cube down to gammapy.background would likely lead to circular
+dependencies, because they rely on the map dataset class in gammapy.cube.
+
+The reflected region background code is not a great fit in gammapy/spectrum or
+anywhere in Gammapy except for gammapy.background, but moving it there does make
+sense if we say that all 1D spectrum code goes in gammapy.spectrum and all
+map-related analysis code goes in gammapy.cube.
```

Comment:
Maybe add a sentence on 0D usage of cumulative statistics that should become part of 1D or 3D Dataset

Reply:
Added this sentence:

> Scalar background statistics,
i.e. for a given region and energy band, can be created via methods on the 1D
and 3D dataset classes.